lib/substitutor.js

```diff
@@ -3,10 +3,11 @@ const Pnr = require("./pnr");
 class Substitutor {
   constructor(testPnrs) {
     this.testPnrsWithTimestamp = testPnrs.sort().map(pnr => {
-      const date = this._parseDateFromPnr(pnr);
+      const dateAndType = this._getDateAndTypeFromPnr(pnr);
```

Comment:
I'd rather have keep `_parseDateFromPnr` and add a new method for getting the type